Directory.Build.targets

```
@@ -65,14 +65,13 @@
   <ItemGroup>
     <!-- Set TargetingPackVersion -->
     <FrameworkReference Update="Microsoft.NETCore.App"
-      Condition=" '$(VSRedistCommonNetCoreSharedFrameworkx6470PackageVersion)'!='' And $(TargetFramework.StartsWith('net7.')) ">
-    <TargetingPackVersion>$(VSRedistCommonNetCoreSharedFrameworkx6470PackageVersion)</TargetingPackVersion>
+      Condition=" '$(Microsoft.NETCore.App.Ref)'!='' And $(TargetFramework.StartsWith('net7.')) ">
+    <TargetingPackVersion>$(Microsoft.NETCore.App.Ref)</TargetingPackVersion>
     </FrameworkReference>
   </ItemGroup>
 
   <PropertyGroup>
-    <!-- Set RuntimeFrameworkVersion to VSRedistCommonNetCoreSharedFrameworkx6470PackageVersion from Core Setup to prevent large SDK cycle -->
-    <RuntimeFrameworkVersion Condition="'$(VSRedistCommonNetCoreSharedFrameworkx6470PackageVersion)'!='' And $(TargetFramework.StartsWith('net7.')) ">$(VSRedistCommonNetCoreSharedFrameworkx6470PackageVersion)</RuntimeFrameworkVersion>
+    <RuntimeFrameworkVersion Condition="'$(MicrosoftNETCoreAppRuntimewinx64PackageVersion)'!='' And $(TargetFramework.StartsWith('net7.')) ">$(VSRedistCommonNetCoreSharedFrameworkx6470PackageVersion)</RuntimeFrameworkVersion>
```

I think this is going to need to be:

```suggestion
    <RuntimeFrameworkVersion Condition="'$(MicrosoftNETCoreAppRuntimewinx64PackageVersion)'!='' And $(TargetFramework.StartsWith('net7.')) ">$(MicrosoftNETCoreAppRuntimewinx64PackageVersion)</RuntimeFrameworkVersion>
```

Before stabilization, `VSRedistCommonNetCoreSharedFrameworkx6470PackageVersion` and `MicrosoftNETCoreAppRuntimewinx64PackageVersion` will have identical versions. After stabilization, `VSRedistCommonNetCoreSharedFrameworkx6470PackageVersion` remains the full version of the runtime, while `MicrosoftNETCoreAppRuntimewinx64PackageVersion` becomes stable. The stable package is what actually gets pushed to NuGet.

This line is telling the .NET SDK that the runtime version it should be referencing should be "7.0.0" or "7.0.1" or whatnot.

---

I missed that one, I originally changed it to MicrosoftNETCoreAppRuntimewinx64PackageVersion (in the test branch), thanks

---

In addition, you'll see https://github.com/dotnet/winforms/blob/release/7.0/global.json#L6 is referencing the non-stable `VSRedistCommonNetCoreSharedFrameworkx6470PackageVersion`. This **is** correct. This is because this entry allows the arcade infra to actually find the runtime layout. When we build a stable runtime for 7.0.0, each time we build it, it will be versioned "7.0.0", but in reality these are different builds. So how do we disambiguate?

For artifacts that are stored in storage accounts, like the runtime layout, we store them underneath the non-stable version number (e.g. `7.0.0-rtm-1234.5/dotnet-runtime-7.0.0-win-x64.zip`). Then we use the non-stable version number in the global.json to look up the correct version.

For NuGet based artifacts, we store the stable versions in separate feeds per-build, and then put the correct feed into the `NuGet.config`.

---

I see, thanks for the knowledge